### PR TITLE
Refactor/DEV-8404: q-figure

### PR DESCRIPTION
**Pull request by hbalenda**
### Changes:
- Move [#94](https://github.com/thegetty/quire-starter-theme/pull/94/files) on `quire-starter-theme` to the monorepo using `git format-patch`. 
- Include and run `prettier` on new partials and shortcode `q-figure`
- Move figure partials into `partials/figures/*`
- Refactor `q-figure-zoom` and `q-figure-group` shortcodes to use `caption` and `label` blocks, setting us up to completely refactor those shortcodes at a later date

### CODE REVIEW

File: themes/default/layouts/partials/figure-caption.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<figcaption class="quire-figure__caption caption">{{ if eq .Site.Params.epub true }}<span class="quire-figure__label">{{ .label }}</span> {{ end }}{{ with .caption }}{{ markdownify . }}{{ end }}{{ with .credit }} <span class="quire-figure__credit">{{ markdownify . }}</span>{{ end }}</figcaption>
```
Comment (mphstudios) [inline themes/default/layouts/partials/figure-caption.html:1]:
```suggestion
<figcaption class="quire-figure__caption caption">
{{- if eq .Site.Params.epub true }}
  <span class="quire-figure__label">{{ .label }}</span>
{{- end }}
{{- with .caption }}
  {{ markdownify . }}
{{- end }}
{{- with .credit }}
  <span class="quire-figure__credit">{{ markdownify . }}</span>
{{- end }}
</figcaption>
```

File: themes/default/layouts/partials/figure-fallback.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  {{- $imageDir = "img" -}}
+{{- end }}
+
+{{- if .src -}}
+
+<img
+  id="{{ .id }}"
+  class="quire-figure__image"
+  src="{{ printf "%s/%s" $imageDir .src | relURL }}"
+  alt="{{ .alt }}" />
```
Comment (mphstudios) [inline themes/default/layouts/partials/figure-fallback.html:14]:
```suggestion
<img
  id="{{ .id }}"
  class="quire-figure__image"
  src="{{ printf "%s/%s" $imageDir .src | relURL }}"
  alt="{{ .alt }}"
/>
```

File: themes/default/layouts/partials/figure-fallback.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+{{- else -}}
+
+<div class="quire-figure__media-fallback">
+<div class="placeholder">
+    <span class="fallback-image">
+      <img src='{{ $imageDir | relURL }}/icons/{{ .mediaType }}.png'/>
+    </span>
+</div>
+</div>
```
Comment (mphstudios) [inline themes/default/layouts/partials/figure-fallback.html:24]:
```suggestion
<div class="quire-figure__media-fallback">
  <div class="placeholder">
    <span class="fallback-image">
      <img src="{{ $imageDir | relURL }}/icons/{{ .mediaType }}.png" />
    </span>
  </div>
</div>
```

File: themes/default/layouts/partials/figure-image.html [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  title="{{ .caption | markdownify | safeHTML }}"
+  class="quire-figure leaflet-outer-wrapper mfp-hide notGet">
+  <div
+    id="js-deepzoom-{{ $now }}"
+    class="quire-deepzoom inset leaflet-inner-wrapper "
+    aria-live="polite"
+    role="application"
+    aria-label="Zoomable image"
+    src="{{ printf "%s/%s" $imageDir .src | relURL }}">
+  </div>
```
Comment (mphstudios) [inline themes/default/layouts/partials/figure-image.html:21]:
```suggestion
  <div
    id="js-deepzoom-{{ $now }}"
    class="quire-deepzoom inset leaflet-inner-wrapper "
    aria-label="Zoomable image"
    aria-live="polite"
    role="application"
    src="{{ printf "%s/%s" $imageDir .src | relURL }}"
  />
```

File: themes/default/layouts/partials/figure-table.html [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  <div
+  id="js-deepzoomtable-{{ $now }}"
+  class="quire-deepzoom inset leaflet-inner-wrapper"
+  aria-live="polite"
+  role="application"
+  aria-label="Zoomable table">
+  <figure class="leaflet-table">{{ readFile $tableSrc | safeHTML }}</figure>
+  </div>
+  <span class="figure-caption visually-hidden">{{ with .caption }}{{ markdownify . }}{{ end }}{{ with .credit }} {{ markdownify . }}{{ end }}</span>
+</figure>
```
Comment (mphstudios) [inline themes/default/layouts/partials/figure-table.html]:
```suggestion
<figure
  id="deepzoom-{{ $now }}"
  title="{{ .caption | markdownify | safeHTML }}"
  class="quire-figure leaflet-outer-wrapper mfp-hide notGet"
>
  <div
    id="js-deepzoomtable-{{ $now }}"
    class="quire-deepzoom inset leaflet-inner-wrapper"
    aria-label="Zoomable table"
    aria-live="polite"
    role="application"
  >
    <figure class="leaflet-table">{{ readFile $tableSrc | safeHTML }}</figure>
  </div>
  <span class="figure-caption visually-hidden">{{ with .caption }}{{ markdownify . }}{{ end }}{{ with .credit }} {{ markdownify . }}{{ end }}</span>
</figure>
```

**Comment by hbalenda** [comment]
@geealbers - What was left to fix on the refactored `q-figure`? The only thing I've noticed so far is that it looks like the styles for the fullscreen controls on `q-figure-zoom` are broken.

**Comment by geealbers** [comment]
@hbalenda I'm not totally sure, it's been too long and my paper to do list about it is mostly about small functionality and variables that I thought should be deprecated/removed, and we could do those in a subsequent PR. 

It should be very close to ready as is, but below are the things I'd do one last test of before merging. And I could do this, but it would have to be next week after I get Rococo out the door.

- All variants of these config.yml parameters: `figureLabelLocation`, `figureModal`, `figureModalIcons`, `figureZoom` (though the last two were things I think we should get rid of)
- Tables
- Audio and Video with and without `src`
- Figures with and without `label`, `caption` and `credit` in various combinations

And I'd look at these things online, in the PDF output, and the EPUB output.

**Comment by hbalenda** [comment]
@geealbers - I have a question about how the `if src, else if id...` logic in `q-figure` works with audio and video. `src` isn't used in the audio/video partials except as the video still placeholder on pdf and epub. Isn't the purpose of the `if src, else if id...` logic in `q-figure` to support the feature that a user should be able to supply `figures.yml` data or attributes defined directly on the shortcode? If so, then we want to move the audio and video partials out of the `if src` block and into an `if media_id` block.

**Comment by geealbers** [comment]
@hbalenda, yes, you're right about the reasoning for the `if src` and `else if id` logic, and you're right that it kind of falls apart with audio and video, I hadn't thought about that before. I would like to keep the ability for users to add figures without adding figures.yml though, as figures.yml can sometimes be a pain and overkill if you're just adding a few figures. 

What if we made it `{{- if or (.Get "src") (.Get "media_id") -}}` so that we didn't have to add a whole other condition just for audio and video?

File: themes/default/layouts/partials/figures/media-types/vimeo.html [hunk 6]
```diff
@@ -0,0 +1,32 @@
+{{- if or (eq .Site.Params.epub true) (eq .Site.Params.pdf true) -}}
+  {{- $fallbackDict := dict "Site" .Site "alt" .alt "id" .id "mediaType" .mediaType "src" .src -}}
+  {{- partial "figures/media-types/fallback.html" $fallbackDict -}}
+
+  <figcaption class="quire-figure__caption caption">
+    <a href="https://youtu.be/{{ .mediaId }}" target="_blank">https://vimeo.com/{{ .mediaId }}</a>
```
[the remark this answers is not in the document]
Comment (hbalenda) [reply themes/default/layouts/partials/figures/media-types/vimeo.html]: Corrected

**Comment by hbalenda** [comment]
@geealbers Thank you so much for the thorough testing! I'm going to write up Jira tasks for and address the existing issues you found in separate PRs to keep this one contained to the shortcode refactor. But I will fix the following outstanding issues before merging this:
- [x] Fix vimeo URL
- [ ] Vertically center label icon
- [ ] Add bold style and right-margin to Epub labels
- [ ] Add line break between fallback URLs and captions on Epub

**Review by geealbers — APPROVED** [review]
All the fixes look good @hbalenda, thanks! Did you already write Jira tickets for the 3 other new issues not related to the refactor?

1. Adding the figure label to the caption in the modal
2. Deprecating `figureModalIcons` and `figureZoom` from config
3. Fixing table insertion/output in EPUB

**Comment by hbalenda** [comment]
@geealbers yep, added you as a watcher to each

**Pull request merged**